src/main/java/htsjdk/samtools/CRAMBAIIndexer.java

```diff
@@ -115,56 +112,60 @@ public CRAMBAIIndexer(final OutputStream output, final SAMFileHeader fileHeader)
      * @param container container to be indexed
      */
     public void processContainer(final Container container, final ValidationStringency validationStringency) {
-        try {
-            if (container == null || container.isEOF()) {
-                return;
-            }
+        if (container == null || container.isEOF()) {
+            return;
+        }
 
-            int sliceIndex = 0;
-            for (final Slice slice : container.slices) {
+        int sliceIndex = 0;
+        for (final Slice slice : container.slices) {
+            slice.containerOffset = container.offset;
+            slice.index = sliceIndex++;
+            if (slice.isMultiref()) {
                 slice.containerOffset = container.offset;
                 slice.index = sliceIndex++;
-                if (slice.isMultiref()) {
-                    final ContainerParser parser = new ContainerParser(indexBuilder.bamHeader);
-                    final Map<Integer, AlignmentSpan> refSet = parser.getReferences(container, validationStringency);
+
+                final Map<Integer, SliceMetadata> metadataMap = slice.getMultiRefAlignmentMetadata(container.header, validationStringency);
+
+                // References must be processed in order, with unmapped last
+                // TODO refactor w/ CRAIIndex.processContainer()

```

Comment:
a fair amount of shared code between these now.  Maybe handle this as part of a general indexing refactor